src/Tribe/Cache.php

```diff
@@ -373,16 +373,23 @@ public function make_key( $components, $prefix = '', $sort = true ) {
 	 * Whether a offset exists.
 	 *
 	 * @since 4.11.0
-	 *
-	 * @link  http://php.net/manual/en/arrayaccess.offsetexists.php
+	 * @since TBD Will check against cache expiration. Previously would give false positive
+	 *            if expiration had passed but was cached recently. Will now consider null not set.
 	 *
 	 * @param mixed $offset An offset to check for.
 	 *
 	 * @return boolean Whether the offset exists in the cache.
+	 *@link  http://php.net/manual/en/arrayaccess.offsetexists.php
+	 *
 	 */
 	#[\ReturnTypeWillChange]
 	public function offsetExists( $offset ) {
-		return isset( $this->non_persistent_keys[ $offset ] );
+		if ( ! isset( $this->non_persistent_keys[ $offset ] ) ) {
+			return false;
+		}
+		$value = $this->get( $offset );
+
+		return $value !== false && $value !== null;
```

Comment:
I think we should definitely consider null as not set, e.g.

```php
$cache['some_id'] = null;
// Null values will no longer pass this check.
if(isset($cache['some_id'])) {
```

This update allows for that, and will also consider expiration time (previously ignored).

Reply:
Regarding expiration time, before this check a situation like this could and was happening...
```php
$cache->set('id', 123, 1); // Set `id` with `123` for 1 second.

// One second later, in same request....

// This would evaluate to true even though `id` expired.
if(isset($cache['id'])) {
  // The expectation would be that the value is `123`, but it would be `false`.
  // False is the default from wp_cache_get() if the value expired.
  $id = $cache['id']; 
}
```

Reply:
```suggestion
		return ! empty( $value );
```
Covers null, false, and empty strings (which should probably also be invalid?)

Reply:
I did consider that, but I wouldn't go that loose. I don't even like considering `false` invalid, I'm only doing so because that's how WP treats their cache getter. I'd lean toward only considering `null` and `false` invalid, because we may legitimately cache various results that are legit but `empty()`. It is very dependent on the specific case.

Reply:
To go on a rant on this... if time/agreement permitted, this is how I'd lean.

I'd consider `false` a legit cache value.
```php
$cache['is_turned_on'] = false; // A bool flag that was set to "off" - legit use case.

if(isset($cache['is_turned_on'])) { // Would be true, because false is a legit value.
```
And internally I'd do some serialization to work around Wordpresses `false` interpretation as an indicator for "cache expired/doesn't exist" from the `wp_cache_get()` func we use. 

e.g. a simplified `Cache` set method to handle bool `false`.
```php
// Cache class
public function set($id, $value) {
  if($value === false) {
    $value = "some_serialized_id";
  }
  wp_cache_set($id, $value);
}
```
Then if we fetch it...
```php
// Cache class
public function get($id) {
  $val = wp_cache_get($id, $value);

  if($val === 'some_serialized_id') {
    return false; // This was the serialized value.
  }

  if($val === false) {
    return null; // This is what should be returned if no cache value exists - null is a standard for "something doesn't exist".
  }

  // All other legit values returned.
  return $val;
}
```

Then `null` would be the only not "legit" value, and I think most people would safely assume that.

Reply:
For me, the biggest question is if we should keep this for the next release so we can run a bunch of QA on that since Cache is used all over the codebase.

Reply:
The `wp_cache_get` function takes a fourth argument `$found` that is set by reference that allows decoupling what is returned from the cache from its found state. Why not leverage that? Any object cache implementation will set and implement that.

Reply:
I like this - didn't see that. Thanks @lucatume .

Reply:
Updated with this line of reasoning - now `false` is a valid value. Do we want to do this? This will require fixing any number of places we looked for `false`.

Also switched the `$default` to `null` in the getter, if no value specified the default will not return a `false` bool. This is to be consistent with `null` being the check for a cache value (where `false` is valid now and would persist).

Reply:
> This will require fixing any number of places we looked for `false`.

I would not change this behaviour of the `Tribe__Cache` API now: `false` might be a valid `wp_cache_get` value, but the `Tribe__Cache` class will treat it as non valid.

Reply:
Then using `$found` is pointless, since the only point was to differentiate `false` stored value vs `false` no stored value.

I'll go back to what I had.

Reply:
The idea was to keep a smart logic that could be dumbed down and "hidden" in the cache API. Now the change is reverted it's fine to leave it like that.